backend/tests/v5/unit/models/modelSettings.test.js

```diff
@@ -136,10 +161,135 @@ const testGetFederations = () => {
+
+const testIsSubModel = () => {
+	describe('Is submodel', () => {
+		test('should return false if not submodel', async () => {
+			const teamspace = 'someTS';
+			const modelId = 'someModel';
+
+			const fn = jest.spyOn(db, 'count').mockResolvedValue(0);
+
+			const res = await Model.isSubModel(teamspace, modelId);
+			expect(res).toEqual(false);
+
+			expect(fn.mock.calls.length).toBe(1);
+			expect(fn.mock.calls[0][0]).toEqual(teamspace);
+			expect(fn.mock.calls[0][1]).toEqual('settings');
+			expect(fn.mock.calls[0][2]).toEqual({ 'subModels.model': modelId });
+		});
+
+		test('should return true if submodel', async () => {
+			const teamspace = 'someTS';
+			const modelId = 'subModel';
+
+			const fn = jest.spyOn(db, 'count').mockResolvedValue(1);
+
+			const res = await Model.isSubModel(teamspace, modelId);
+			expect(res).toEqual(true);
+
+			expect(fn.mock.calls.length).toBe(1);
+			expect(fn.mock.calls[0][0]).toEqual(teamspace);
+			expect(fn.mock.calls[0][1]).toEqual('settings');
+			expect(fn.mock.calls[0][2]).toEqual({ 'subModels.model': modelId });
+		});
+	});
+};
+
+const testRemoveModelCollections = () => {
+	describe('Remove model collections', () => {
+		test('should succeed', async () => {
+			const modelId = 'someModel';
+			const collectionList = [
+				{ name: `${modelId}.collA` },
+				{ name: `${modelId}.collB` },
+				{ name: 'otherModel.collA' },
+				{ name: 'otherModel.collB' },
+			];
+
+			const fnList = jest.spyOn(db, 'listCollections').mockResolvedValue(collectionList);
+			const fnDrop = jest.spyOn(db, 'dropCollection').mockResolvedValue(true);
+
+			const teamspace = 'someTS';
+			const res = await Model.removeModelCollections(teamspace, modelId);
+
+			expect(res).toEqual([true, true]);
+
+			expect(fnList.mock.calls.length).toBe(1);
+			expect(fnList.mock.calls[0][0]).toEqual(teamspace);
+
+			expect(fnDrop.mock.calls.length).toBe(2);
+			expect(fnDrop.mock.calls[0][0]).toEqual(teamspace);
+			expect(fnDrop.mock.calls[0][1]).toEqual(collectionList[0]);
+			expect(fnDrop.mock.calls[1][0]).toEqual(teamspace);
+			expect(fnDrop.mock.calls[1][1]).toEqual(collectionList[1]);
+		});
+	});
+};
+
+const testAddModel = () => {
+	describe('Add model', () => {
+		test('should return inserted ID on success', async () => {
+			const newContainerId = 'newContainerId';
+			const expectedData = { insertedId: newContainerId };
+
+			const fn = jest.spyOn(db, 'insertOne').mockResolvedValue(expectedData);
+
+			const teamspace = 'someTS';
+			const res = await Model.addModel(teamspace, {});
+			expect(res).toEqual(newContainerId);
+			expect(fn.mock.calls.length).toBe(1);
+			expect(fn.mock.calls[0][0]).toEqual(teamspace);
+			expect(fn.mock.calls[0][1]).toEqual('settings');
+			expect(fn.mock.calls[0][2]).toHaveProperty('_id');
+			expect(isUUIDString(fn.mock.calls[0][2]._id));
+		});
+	});
+};
+
+const testDeleteModel = () => {
+	describe('Delete model', () => {
+		test('should return deleted count on success', async () => {
```

Comment:
it doesn't return deleted count, so I think we should rename this test title a bit